Request to add the profiles for the FLSun QQS-Pro and Q5 Delta printers.

Foxies-CSTL opened the pull request:
Hello team,

I would like to submit the printing profiles for the Deltas QQS & Q5 STOCK printers.
I realized, tested and validated (*) with the help of other users, several of these profiles.
Here is what it contains:
- 0.4 nozzle
  0.08 DETAIL
  0.16 QUALITY (*) & SPEED (*)
  0.20 QUALITY (*) & SPEED (*)
  0.24 DRAFT
  0.30 SPEED (*)
  0.15 VASE NORMAL & SPEED + PLA (*)
- 0.6 nozzle
  0.16 DETAIL
  0.20 DETAIL
  0.30 QUALITY (*)
  0.35 SPEED
  0.40 DRAFT
-Generic PLA (*) / PETG (*) / SPLA (*) / FLEX (*) / ABS filaments

I tried to respect your structure linked to the MINI (Bowden) and I was inspired by certain other profiles to adapt them to our printers.
I hope I haven't made too many mistakes. I will make them evolve according to the feedback from more experienced users.
This is a good basis to start with the FLSun QQS & Q5 in 0.4 PLA / PETG nozzle, QUALITY, SPEED and VASE mode.

Thank you for your attention and happy Holidays.

Additional Info:
Pack_Profiles: 
[0.0.1.zip](https://github.com/prusa3d/PrusaSlicer/files/5741954/0.0.1.zip)

I realized the Marlin firmware for QQS-Pro and I assure the maintenance of the corrections.
https://github.com/Foxies-CSTL/Marlin_2.0.x/tree/FLSUN_QQS-PRO-MULTI/Marlin/Firmwares
I belong to the FaceBook group (> 5000 users)
https://www.facebook.com/groups/120961628750040/about

PS-BenchyFamily
![PS-BenchyFamily](https://user-images.githubusercontent.com/24864147/103134697-65721e00-46b3-11eb-9b51-8defa36e509e.png)

PS-Tests_Plateau-PLA
![PS-Tests_Plateau-PLA2](https://user-images.githubusercontent.com/24864147/103134712-7327a380-46b3-11eb-9f92-c471e807fab7.png)
PS-Test_Accel-Jerk
![PS-Test_Accel-Jerk_PLA2](https://user-images.githubusercontent.com/24864147/103134717-7ae74800-46b3-11eb-9d8f-5ec03b43fdd4.jpg)
![PS-Test_Accel-Jerk_PLA](https://user-images.githubusercontent.com/24864147/103134721-7e7acf00-46b3-11eb-9390-f9b6f3570224.jpg)
PS-Test_0.3NozzlePLA
![PS-Test_PLA0 3](https://user-images.githubusercontent.com/24864147/103134723-83d81980-46b3-11eb-9677-4d3f6ea818f0.jpg)
![PS-Test_PLA0 3-2](https://user-images.githubusercontent.com/24864147/103134728-88043700-46b3-11eb-9294-7971c19d6ce4.jpg)

TestPrint-CompleteIndividualObjet
![TestPrint-CompleteIndividualObjet](https://user-images.githubusercontent.com/24864147/103134739-9a7e7080-46b3-11eb-862b-8b4ce1e4851c.png)
![IMG_1993](https://user-images.githubusercontent.com/24864147/103134830-5cce1780-46b4-11eb-8339-5a421915205a.JPG)

rtyr commented:
Thank you. I will look into it.

rtyr commented:
The bundle looks good overall, I only fixed some small issues in the structure (incorrect `seam_position` key in printer section and incorrect use of inheritance with `*PLA_VASE_flsunq*`). 

I have some objections/questions though :-).

- FLSun Q5 (and 0.6 nozzle variant) printer profile will certainly not work. It is using absolute positioning for the extruder ("Use relative E distances" parameter is disabled) in the combination with custom G92 E0 command before each layer. It will lead to this https://github.com/prusa3d/PrusaSlicer/issues/4327. QQS Pro is OK in this regard. Probably just small unwanted misconfiguration in the ini file. I will enable "Use relative E distances".

- I am not a fan of the bed texture. There are some visible artefacts and the checkerbox pattern is causing various moire effects on my monitor. I made STL model for the bed and added different texture. I hope it is OK with you.

- I would not reccommend to enable "Detect thin walls" parameter by default. It can lead to unwanted issues like https://github.com/prusa3d/PrusaSlicer/issues/227.

- You have the Ironing enabled by default in "Quality" profiles, is it intended? I am not sure that users will expect that option to be enabled by default and it also increases print time significantly.

- I personally don't like the "Only retract when crossing perimeters" option enabled by default, but I will leave it to your decision. The reason I don't like it enabled by default is that it is usually very confusing for the new users (ugly looking infill with lot of strings). It also disables z-lift along with the retractions and this feature can lead to missing material after long nonretracted travel moves.

- what is the "0.20mm linear advance" profile meant for? We had such profile back in the day when we wanted to highlight the benefits of the linear advance feature. I can see that LA is not used in your profiles and I am not sure if it is enabled in the stock FLSUN firmware.

Proposed bed stl and texture:
![bed](https://user-images.githubusercontent.com/36745189/103279644-85188780-49ce-11eb-8d30-e07582c7d005.jpg)

Foxies-CSTL commented:
Hi @rtyr , 
Thanks for this report.

> The bundle looks good overall, I only fixed some small issues in the structure (incorrect `seam_position` key in printer section and incorrect use of inheritance with `*PLA_VASE_flsunq*`).

Ok, I wanted this artifact to link to the Vase profile. Users love to make vase on Deltas ;-)

> I have some objections/questions though :-).
;-) Nice.
 
>     * FLSun Q5 (and 0.6 nozzle variant) printer profile will certainly not work. It is using absolute positioning for the extruder ("Use relative E distances" parameter is disabled) in the combination with custom G92 E0 command before each layer. It will lead to this #4327. QQS Pro is OK in this regard. Probably just small unwanted misconfiguration in the ini file. I will enable "Use relative E distances".

Yes, I have already corrected on my home version (For all "Relative E distance" enable) and add in startGCode a 90° arc purge line for Q5.

>     * I am not a fan of the bed texture. There are some visible artefacts and the checkerbox pattern is causing various moire effects on my monitor. I made STL model for the bed and added different texture. I hope it is OK with you.

Great, it was a user who suggested it to me so I integrated it but your suggestion also pleases me. Okay for yours ;-)
 
>     * I would not reccommend to enable "Detect thin walls" parameter by default. It can lead to unwanted issues like #227.
> 
>     * You have the Ironing enabled by default in "Quality" profiles, is it intended? I am not sure that users will expect that option to be enabled by default and it also increases print time significantly.
> 
Ok for "Detect.." and "Ironing", I removed them too but I also have feedback users for "Only Infill where Need" desactived .
What do you think for that?

>     * I personally don't like the "Only retract when crossing perimeters" option enabled by default, but I will leave it to your decision. The reason I don't like it enabled by default is that it is usually very confusing for the new users (ugly looking infill with lot of strings). It also disables z-lift along with the retractions and this feature can lead to missing material after long nonretracted travel moves.
> 
Ok, I hadn't seen the link with this newbie logic so we deactivate by default "Only Retract..." but we leave Z-lift and retraction.

>     * what is the "0.20mm linear advance" profile meant for? We had such profile back in the day when we wanted to highlight the benefits of the linear advance feature. I can see that LA is not used in your profiles and I am not sure if it is enabled in the stock FLSUN firmware.
> 
For LA, I copied this part for testing (we can remove it) because I activated it in the Marlin firmware that I distribute. Some connoisseurs use it and that is why it is commented on the GCode Start in "Common filament preset". This is to give an example of employment. Otherwise the Stock FLSun firmware (Clone Repetier / Marlin) doesn't use it.
 
> Proposed bed stl and texture:
> ![bed](https://user-images.githubusercontent.com/36745189/103279644-85188780-49ce-11eb-8d30-e07582c7d005.jpg)

Thanks, I'll integrate it to replace the old one ;-)
Tomorrow because it is late at home ;-), I will delta the last modifications to complete this PR. Thanks for the help and happy New Year's Eve !!

rtyr commented:
> Ok, I wanted this artifact to link to the Vase profile. Users love to make vase on Deltas ;-)

Sure, but it is not possible to inherit filament setting to the print setting. You can run `prusa-slicer-console.exe --loglevel=5` to check for errors like this.

> Ok for "Detect.." and "Ironing", I removed them too but I also have feedback users for "Only Infill where Need" desactived .
> What do you think for that?

I would certainly leave "Only infill where needed" disabled (it seems already disabled in this pull request).

> Ok, I hadn't seen the link with this newbie logic so we deactivate by default "Only Retract..." but we leave Z-lift and retraction.

Of course z-lift and retraction should be enabled :-). I was just refering to the side effect of the "only retract when crossing perimeters" (it removes some retracts, but as a side effect, z-lifts related to those retracts are also removed). As I mentioned before, I will leave this setting to you, but I would prefer it disabled by default (or maybe enabled just for SPEED profiles and not for QUALITY profiles?) to prevent FB posts like "PrusaSlicer prints ugly infill" :-). 

> For LA, I copied this part for testing (we can remove it) because I activated it in the Marlin firmware that I distribute. Some connoisseurs use it and that is why it is commented on the GCode Start in "Common filament preset". This is to give an example of employment. Otherwise the Stock FLSun firmware (Clone Repetier / Marlin) doesn't use it.

OK, no problem. 

> Thanks, I'll integrate it to replace the old one ;-)

Well, I have to send it to you first :-) (I have it only locally). It is attached to this post.
[FLSun_bed.zip](https://github.com/prusa3d/PrusaSlicer/files/5756232/FLSun_bed.zip)

Foxies-CSTL commented:
Thank you for your support and .........

> I would prefer it disabled by default (or maybe enabled just for SPEED profiles and not for QUALITY profiles?) to prevent FB posts like "PrusaSlicer prints ugly infill" :-).
 I've already had user feedback for this :-D

....... Best Wishes to the PrusaSlicer team !!

rtyr commented:
Commited to the master with some minor changes and different thumbnails. The bundle will be included in the next PrusaSlicer release. 3rd party profiles can be updated on-line. Closing.

rtyr commented:
Thank you.

Foxies-CSTL commented:
> 
> 
> Commited to the master with some minor changes and different thumbnails. The bundle will be included in the next PrusaSlicer release. 3rd party profiles can be updated on-line. Closing.

Great, thanks for all.
